napari/utils/io.py

```diff
@@ -1,16 +1,85 @@
 import os
+import re
 
 from glob import glob
 from pathlib import Path
 
 import numpy as np
-from skimage import io
-from skimage.io.collection import alphanumeric_key
 
 from dask import delayed
 from dask import array as da
 
 
+def imsave(filename: str, data: np.ndarray):
+    """custom imaplementation of imread to avoid skimage dependecy"""
+    ext = os.path.splitext(filename)[1]
+    if ext in [".tif", "tiff"]:
+        import tifffile
+
+        tifffile.imsave(filename, data)
+    else:
+        import imageio
+
+        imageio.imsave(filename, data)
+
+
+def imread(filename: str):
+    """custom imaplementation of imread to avoid skimage dependecy"""
+    ext = os.path.splitext(filename)[1]
+    if ext in [".tif", "tiff", ".lsm"]:
+        import tifffile
+
+        image = tifffile.imread(filename)
+    else:
+        import imageio
+
+        image = imageio.imread(filename)
+
+    if not hasattr(image, 'ndim'):
+        return image
+
+    if image.ndim > 2:
+        if image.shape[-1] not in (3, 4) and image.shape[-3] in (3, 4):
+            image = np.swapaxes(image, -1, -3)
+            image = np.swapaxes(image, -2, -3)
```

Comment:
Honestly, I would simply remove this logic. It is something archaic that we do in skimage and we will probably remove it in 1.0.

Reply:
It is no so archaic as you think. I work with files that need this, but `tifffile` allow to do it better. Tifffile image format in metadata store axes order and do not force this order (The data with which I work has axes order ZCYX). 

In my program (PartSeg) I have separated module to read image with metadata (based on tifffile), which is prepared to extract to separate package. You can see logic to ensure axes order here https://github.com/4DNucleome/PartSeg/blob/master/package/PartSegImage/image_reader.py#L97

Reply:
@Czaki what I mean is that napari lets you do `channel_axis=1` for your ZCYX image, so the channels don't need to be at the final position. The *assumption* of the position of the channel axis is not needed. Do you agree?

Reply:
Hm. It depends. If one use napari only as viewer then it is not needed. If One would like to test general data processing algorithm (which does not need to be changed when microscope is changed) then order of axes is important

Reply:
It *is* important, but my point is that it should be a parameter, not hardcoded at the end.

Reply:
@Czaki I still would like to remove this whole clause. It is brittle and results in inconsistencies. For example, the famous `mitosis.tif` dataset has order TZCYX, *but* only two channels, so this code will not place the channel axis at the end even though that is the intention. Some other datasets might have a single channel but still have that dimension, or 5 channels. Another dataset might be simply ZYX and single-channel, but Z might only have length 4, and this code will incorrectly transpose that axis.

As you point out, the correct way to do this is with metadata, not with guesses about the shapes of things.